cmd/cosign/cli/attach/sig.go

```diff
@@ -98,20 +98,27 @@ func SignatureCmd(ctx context.Context, regOpts options.RegistryOptions, sigRef,
 		}
 	}
 	tsBundle := bundle.TimestampToRFC3161Timestamp(timeStampedSig)
-
-	if rekorBundleRef != "" {
-		rekorBundleByte, err := os.ReadFile(filepath.Clean(rekorBundleRef))
+	if rekorResponseRef != "" {
+		rekorResponseByte, err := os.ReadFile(filepath.Clean(rekorResponseRef))
 		if err != nil {
 			return err
 		}
 
 		var localCosignPayload cosign.LocalSignedPayload
-		err = json.Unmarshal(rekorBundleByte, &localCosignPayload)
+		err = json.Unmarshal(rekorResponseByte, &localCosignPayload)
+		if err != nil {
```

Comment:
We wouldn't want to return on the first error, since the input may be one of two structures. We'll want something like:

```
var localCosignPayload cosign.LocalSignedPayload
if err := json.Unmarshal(rekorResponseByte, &localCosignPayload); err == nil {
  rekorBundle = localCosignPayload.Bundle
} else {
  err := json.Unmarshal(rekorResponseByte, rekorBundle)
  if err != nil {
    return fmt.Errorf("unable to parse Rekor response to attach to image")
  }
}
```

This also removes the need for the extra `duplicateLocalCosignPayload` variable since you can use `rekorBundle` directly.

Reply:
Comment still relevant. Line 110 would return immediately if there was an error.

Reply:
@viveksahu26 Just bumping